### PR TITLE
Ends the Security Sprite War FOREVER

### LINKED ISSUE #17755: Security Sprites are causing constant arguments and reverts and counter PRs, and are distracting from our maintainer's precious time
We should look into a way to have ALL security sprites without having to have maintainers waste time arguing over the color of a jumpsuit.

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
rscadd: Security uniforms are now subcontracted to other companies.
/:cl:
Fixes #17755

### CODE REVIEW

**Comment by TechnoAlchemisto** [comment]
conflicts?!??!!!!

**Comment by PKPenguin321** [comment]
HE ACTUALLY DID IT THE ABSOLUTE MADMAN

**Comment by Iamgoofball** [comment]
This is a fix, IDK what you're doing

**Comment by PKPenguin321** [comment]
>INB4 SOME KIND OF GLITCH THAT MAKES SECS CLOTHES SPAWN IN NULLSPACE LEAVING THEM NAKED AND EVENTUALLY CRASHING THE SERVER

**Comment by Iamgoofball** [comment]
its not a feature its a bugfix for #17755 @KorPhaeron

**Comment by Bawhoppen** [comment]
a final solution :+1: :nail_care:

**Comment by Steelpoint** [comment]
Please explain what this does for us illiterate people.

**Comment by TechnoAlchemisto** [comment]
random uniforms

**Comment by Steelpoint** [comment]
Aww no.

I feel this would be a good opportunity to give the HoS something of positive benefit that not only allows him/her to influence the appearance of the department, but also be a nice bonus for playing one of the most stressful roles on the station. 

The appearance should be something the HoS selects as a config option.

**Comment by Jordie0608** [comment]
> Subsystem for selecting security sprites

You've got to be kidding me

**Comment by Bawhoppen** [comment]
this reminds me how I wanted to do custom loadouts

File: code/controllers/subsystem/alt_sprites.dm [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	var/list/sec_themes = list(/datum/sprite_theme/oldsec, /datum/sprite_theme/greysec, /datum/sprite_theme/copsec, /datum/sprite_theme/corpsec, /datum/sprite_theme/redsec)
+	var/help_me = pick(sec_themes)
+	var/datum/sprite_theme/S = new help_me
+	for(var/O in S.overrides)
+		sprite_overrides += new O
+	for(var/datum/sprite_override/OR in sprite_overrides)
+		OR.adjust_all_sprites()
+
+/datum/sprite_theme
+	var/name = "redtube"
```
Comment (RemieRichards) [inline code/controllers/subsystem/alt_sprites.dm]: Nope, not happening.

File: code/controllers/subsystem/alt_sprites.dm [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		A.update_icon()
+
+/datum/sprite_override/proc/adjust_a_sprite(var/obj/item/A)
+	A.icon_state = initial(chosen.icon_state)
+	A.item_state = initial(chosen.item_state)
+	A.item_color = initial(chosen.item_color)
+	A.update_icon()
+
+/datum/sprite_alt
+	var/name = "http://www."
```
Comment (RemieRichards) [inline code/controllers/subsystem/alt_sprites.dm]: N

File: code/controllers/subsystem/alt_sprites.dm [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	var/icon_state = "porn"
```
Comment (RemieRichards) [inline code/controllers/subsystem/alt_sprites.dm]: O

File: code/controllers/subsystem/alt_sprites.dm [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	var/item_state = "hub"
```
Comment (RemieRichards) [inline code/controllers/subsystem/alt_sprites.dm]: P

File: code/controllers/subsystem/alt_sprites.dm [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	var/item_color = ".com"
```
Comment (RemieRichards) [inline code/controllers/subsystem/alt_sprites.dm]: E

**Comment by kevinz000** [comment]
nope

**Comment by Man-Shroom-** [comment]
SO I CAN WORK LONGER

**Comment by MMMiracles** [comment]
goof please, dont let your memes overwhelm the potential good this PR can do.

sacrifice the memes just this once for the greater good that is ending the sprite wars

**Comment by Bawhoppen** [comment]
I'll end it personally if he doesn't
also :snowflake: anyways

**Comment by Iamgoofball** [comment]
@Jordie0608 It's designed for more than security sprites. Anytime we get an alternate sprite, we can hook it up to this system for the game to randomize every round.

File: code/controllers/subsystem/alt_sprites.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/controllers/subsystem/alt_sprites.dm]: i'll change it but I really dont see the problem here, it's just a debug name

**Comment by Steelpoint** [comment]
But being serious if this is going to be expanded into multiple departments then the head of that department really needs control over the theme/colour/item used.

As I've been saying since a year ago when this first sprite debate started, but this kind of change would be a fantastic opportunity to give heads of staff a boon to playing the role WITHOUT it being a item that could be seen as overpowered, broken or requiring a lot of balancing otherwise. 

Its pure fluff and its a guaranteed 100% win/win for all parties involved. No drawbacks.

**Comment by Iamgoofball** [comment]
perhaps someone can code a tool down the line to change the randomized one post round-start

for now it stays randomized, don't bikeshed kids

**Comment by Iamgoofball** [comment]
@Bawhoppen its a fix, not a feature freeze

**Comment by GunHog** [comment]
This is not a fix; You know that this is a feature, and you are not getting this merged without a maintainer being high, drunk, or bribed. Unfortunately, memes DO happen...

Anyway, I do not think this should be approved without giving the head of staff the power to choose. Another worry of mine is that expanding this "fix" to other departments will start sprite wars there as well.

**Comment by Niknakflak** [comment]
This should be instantly denied due to all the porn shit
Goofball, you're retarded

**Comment by ike709** [comment]
Besides the porn, this is a great PR. My only complaint is that there should be a character setting for the HoS (and any future Head this system applies to) to select their preferred department uniform.

File: code/controllers/subsystem/alt_sprites.dm (hunk 2 above)
Comment (PKPenguin321) [inline code/controllers/subsystem/alt_sprites.dm]: The problem is that this meme is too dank to merge

**Comment by Iamgoofball** [comment]
@ike709 no more fucking preferences jesus christ

it'll be an item if we even add something for the HoS to set this shit

**Comment by Man-Shroom-** [comment]
How would that work as an item even?

Wouldn't that let anyone just steal it and make sec randomly switch colors forever?

**Comment by Iamgoofball** [comment]
it´d either be a wallmounted machine or a handheld switcher with either 1 or infinite uses

the randomized outfit would still happen, you could just change it with the item or the wall mounter

not my problem to code however

**Comment by Zilenan91** [comment]
So what you're saying is that this can remove outfit association from people and makes it harder for new players to differentiate between roles easily?

Alternate sprites should never ever be a thing for anything but backpacks, certainly not for something like a jumpsuit. Newer players are already overwhelmed as fuck just by the controls, forcing them to see the difference between a sec officer and another sec officer that looks wildly different, with aesthetics that change on a literally random basis is just adding on to that pile and will make the game less accessible and drive people away.

**Comment by TheNightingale** [comment]
Why not just add the new variants to the security wardrobes?

**Comment by jughu** [comment]
Lawclothesvendor the autistic subtype of sec vendors should replace wardrobe :^)

**Comment by Zilenan91** [comment]
Put them in the wardrobe, by all means, if you want them in the game, but forcing visual differences on things that are designed to be visually distinctive is beyond terrible design.

**Comment by Iamgoofball** [comment]
All 5 outfits very clearly look like a security officer/corporate goon/redshirt/beat cop, don't worry

**Comment by Zilenan91** [comment]
It doesn't matter that they do, it only matters that new players will be able to instantly identify these people as security officers. If the look changes every single round they cannot do this and they will be alienated.

**Comment by ChangelingRain** [comment]
tbh i should rename this pr to `MAKES THE SEC SPRITE WAR WORSE` so it's accurate

**Comment by Iamgoofball** [comment]
why are you acting like new players are too dumb to think "huh the guy with the helmet, armor, and red outfits might be a security guy"

File: code/controllers/subsystem/alt_sprites.dm (hunk 1 above)
Comment (Xhuis) [inline code/controllers/subsystem/alt_sprites.dm]: LOL

**Comment by Zilenan91** [comment]
The fact that they need to take the time to think that is the problem. If you design a game, you want something as important as identifying a security officer to be an instant process without thought. If it changes every round it will make people pause and be confused, and that confusion will make them get dunked on. This isn't a problem for older players, of course, since they've seen everything and mostly know what's going on with the server codewise, and this entire feature is designed for them, not for everyone.

**Comment by jughu** [comment]
This is a game of pure thinking

**Comment by Zilenan91** [comment]
Yes, but if you're trying to make such a radical change to visual design as this you need to know how visual design works.

Also I could make a strong argument that it's not. A lot of additions to the codebase are designed to be as easy and accessible as possible. There's no real planning that needs to go into anything but building, and it's only a thing for that because it takes so long.

**Comment by jughu** [comment]
ITS EITHER ROBUST MAKING SMART MOVES OR THINKING MAKING YOU PLAN SO ITS A THINKING GAME!!!!

**Comment by Zilenan91** [comment]
Thinking games have to be slow to let players plan, /tg/ is the opposite of slow. You could probably get away with this system if /tg/ was Lifeweb speed but it's not so...

**Comment by Iamgoofball** [comment]
all of these outfits are prior security outfits

**Comment by Zilenan91** [comment]
Yes, and older players know this, new players do not and don't care.

**Comment by jughu** [comment]
Thinking games dont always need to be slow scrub

**Comment by xxalpha** [comment]
You need a dedicated SS for this.

**Comment by WJohn** [comment]
Like others have said, if you want fashion options, leave them in the wardrobe. What you spawn with should always be the same.

As for sprite quality, they look fine, but the default red is very iconic, now. Keep those as default.

**Comment by Iamgoofball** [comment]
@WJohn Nope.

It's going to be randomized, and the HoS will have a console in his office to change it.

**Comment by Iamgoofball** [comment]
THIS IS A COMPROMISE THOUGH

File: code/controllers/subsystem/alt_sprites.dm [hunk 6]
```diff
@@ -0,0 +1,57 @@
+var/datum/subsystem/altsprites/SSaltsprites
+
+/datum/subsystem/altsprites
+	name = "Alternate Sprites"
+	priority = -420
+	wait = 69
+
```
Comment (ShadowDeath6) [inline code/controllers/subsystem/alt_sprites.dm]: Nice

**Comment by CosmicScientist** [comment]
You must compromise until you are at our beck and call without a say in the matter.

**Comment by optimumtact** [comment]
Reopen when the freeze is ended thank you.

**Comment by Iamgoofball** [comment]
Excuse me this is a bugfix

**Comment by CosmicScientist** [comment]
We'll be watching, goof.

**Pull request closed without merging**